### PR TITLE
Fixes Decode-qr

### PR DESCRIPTION
**Pull request by aashna27**
Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] ask `@publiclab/is-reviewers` for help, in a comment below

image

image

image

### CODE REVIEW

**Comment by aashna27** [comment]
@publiclab/is-reviewers

**Comment by aashna27** [comment]
@jywarren

File: src/modules/DecodeQr/Module.js [hunk 1]
```diff
@@ -18,16 +21,25 @@ module.exports = function DoNothing(options,UI) {
...
       var h = pixels.shape[1];
-      var decoded = jsQR.decodeQRFromImage(pixels.data,w,h);
+      var decoded = jsQR(pixels.data,w,h);
 
-      // This output is accessible to Image Sequencer
-      step.output = input;
-      step.output.data = decoded;
 
       // Tell Image Sequencer that this step is complete
-      callback();
```
Comment (HarshKhandeparkar) [inline src/modules/DecodeQr/Module.js]: Shouldn't the callback be called at the end?

File: src/modules/DecodeQr/Module.js [hunk 2]
```diff
@@ -18,16 +21,25 @@ module.exports = function DoNothing(options,UI) {
...
+      step.output = {
+        src: datauri,
+        format: mimetype
+      };
+    }
+    return require('../_nomodule/PixelManipulation.js')(input, {
+      output: output,
+      format: input.format,
+      image: options.image,
+      callback: callback
```
[the remark this answers is not in the document]
Comment (HarshKhandeparkar) [reply src/modules/DecodeQr/Module.js]: Sorry I did not see that `pixelManipulation` part. My bad.

[the remark this answers is not in the document]
Comment (aashna27) [reply src/modules/DecodeQr/Module.js]: No that's okay.

**Comment by aashna27** [comment]
thoughts !!  @publiclab/is-reviewers

**Comment by rexagod** [comment]
Hey could you add the issue number at the top? Thanks.

**Comment by aashna27** [comment]
> 
> 
> Hey could you add the issue number at the top? Thanks.

No issue has been created for this.

**Comment by rexagod** [comment]
Okay. Would to mind stating that at the top in place of `Fixes #[Add issue number here.]`? Just trying to keep things tidy here! That and a rebase and then I think we can merge this!

**Comment by aashna27** [comment]
> 
> 
> Okay. Would to mind stating that at the top in place of `Fixes #[Add issue number here.]`? Just trying to keep things tidy here! That and a rebase and then I think we can merge this!

Yeah sure and since a few prs are ready to merge, i ll rebase this one once they get merged , to prevent rebasing again.

**Comment by jywarren** [comment]
Great. I think good to rebase now?

**Comment by aashna27** [comment]
@jywarren I have rebased, please have a look.

**Comment by aashna27** [comment]
@jywarren

**Comment by jywarren** [comment]
👍 great work!

**Pull request merged**